### PR TITLE
V0.1

### PR DESCRIPTION
**Pull request by therealmistericraft**
Even if the branch name is V1.0, I decided to merge the contents directly to master, even if the bot does nothing. I got to know what branches are for, so everything working will be in master, everything I work on in the new branch "update" (or similar) and releases the bot runs will appear as a new release.

In this pull request, I did the basic structure, so the bot can technically run, but does not know anything it should.

# Detailed changes:
## Folder structure:
A release-zip contains in the first-level path beside the readme and license- (or other GitHub-related-) files two folders, `data` and `resources`. In the former, files the bot does not need are stored in, like logos or tutorials. In the latter, the bot resources like source code ("src") or user data ("data/usr") or other json´s ("data/...") are stored in.
## commands:
I wrote code to change the language for the guild ("setlanguage"), to change the prefix ("setprefix") and a test command.
## events/listener:
I copied the `on_message` event from [MisteriBot](https://www.github.com/therealmistericraft/MisteriBot) and removed unneccessary code and optimized the code to work with json format which is used in schoolbot instead of txt by the MisteriBot.
## other:
I initialized all variables at the beginning (Part #4) and wrote a documentation which shows the developer how to use variables, imports and similar.

### CODE REVIEW

**Review by joseywoermann — CHANGES_REQUESTED** [review]
I suggest to you that you use `indents=4` in yout `json.dump()`-method, so that the file "lang.json" looks better.

File: resources/src/schoolbot.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    if message.content == "prefix":
+        embed = discord.Embed(title="Das Prefix dieses Servers ist ``"+prefix+"``.", colour=discord.Colour.blue())
+        await message.channel.send(content=None, embed=embed)
+    await client.process_commands(message)
+
+#6.2 on_guild_join: Send tutorial to guild owner how to set up the bot
+@client.event
+async def on_guild_join(guild):
+    owner = guild.owner
+    await owner.send(content=None, embed=discord.Embed(title="Thanks for adding schoolbot to your Server!", description="To get started, you have to set up your language. Before this, nothing will work.", colour=discord.Colour.orange()).add_field(name="Start", value="Type `§setlanguage` followed by your language in small letters (e.g. `§setlanguage german`) in a channel the Bot is allowed to read and write in.", inline=False).add_field(name="Trubleshooting", value="I am a junior developer and do not know everything, so the language choice is not reliable nor stable. Please contact me by sending a private message to the bot."))
```
Comment (joseywoermann) [inline resources/src/schoolbot.py:117]: It is called "troubleshooting", not "trubleshooting"

File: resources/src/schoolbot.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+#8 Commands
+
+#8.1 setlanguage
+@client.command(aliases = ['changelanguage', 'cl', 'language'])
+async def setlanguage(ctx, pLanguage):
+    global langs
+    if pLanguage == "german" or pLanguage == "english":
+        langs[str(ctx.guild.id)] = pLanguage
+        with open("../data/usr/lang.json", "w") as langfile:
+            json.dump(langs, langfile)
```
Comment (joseywoermann) [inline resources/src/schoolbot.py:145]: For more beautiful code, do `json.dump(langs, langfile, indents=4)`

Comment (joseywoermann) [reply resources/src/schoolbot.py:145]: Also in line 156

File: resources/src/schoolbot.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        await ctx.send("Language has been set up successfully! Your language: `"+pLanguage+"`")
+    else:
+        await ctx.send("Your language is not available. Availavle languages are: `german`, `english`")
```
Comment (joseywoermann) [inline resources/src/schoolbot.py:148]: Spelling mistake: "available", not "availavle"

File: resources/src/schoolbot.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    global lang
+    await ctx.send(lang["0"])
+
+
+
+
+
+#9 Token
+
+client.run("Nzg5MTY3OTg1NDQ0NjUxMDA4.X9uH9Q.lPOyuOJJbJQJdqu1JISDxiLFliM")
```
Comment (joseywoermann) [inline resources/src/schoolbot.py:171]: Don't publish tokens.

File: resources/src/schoolbot.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    global msg_eng
+    if message.guild:
+        if not message.content == '§setlanguage':
+            if langs[str(message.guild.id)]:
+                if langs[str(message.guild.id)] == "german":
+                    lang = msg_ger
+                if langs[str(message.guild.id)] == "english":
+                    lang = msg_eng
+            elif message.content.startswith == prefix:
+                await message.channel.send(content=None, embed=discord.Embed(title='Error 001', description='There is no language set up for this guild/server. Please contact server owner, which can set up the language with `*setlanguage`.', colour=discord.Colour.red()))
```
Comment (joseywoermann) [inline resources/src/schoolbot.py:98]: Not really mistake, but you could say "...contact the server's owner..."

**Comment by therealmistericraft** [comment]
Hey, @joseywoermann !
Thank you for your feedback.
I realized the optical issue, but do not know how to change it. May you do the changes directly?
Good evening, Jan

**Comment by joseywoermann** [comment]
@therealmistericraft I don't think I have the required permissions to edit these files and as far as I know, one can not make forks of pull requests.

~joseywoermann

**Review by joseywoermann — COMMENTED** [review]
> @therealmistericraft I don't think I have the required permissions to edit these files and as far as I know, one can not make forks of pull requests.
> 
> ~joseywoermann

Oh wait, I can just fork the branch

**Pull request merged**